Code/Tools/SceneAPI/SceneCore/Containers/Scene.cpp

```diff
@@ -42,6 +42,12 @@ namespace AZ
                 return m_sourceFilename;
             }
 
+            const AZStd::string Scene::GetSourceExtension() const
+            {
+                const AZ::IO::PathView sourcePath(m_sourceFilename);
+                return sourcePath.Extension().String();
+            }
```

Comment:
How about returning a `string_view`?
```suggestion
            AZStd::string_view Scene::GetSourceExtension() const
            {
                const AZ::IO::PathView sourcePath(m_sourceFilename);
                return sourcePath.Extension();
            }
```

Reply:
This can also return an `AZ::IO::PathView`.


The user can then use [AZ::IO::PathView::Native](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L153-L158) to retrieve a reference to the string_view from it, if need be.

Reply:
Wouldn't that be creating a view on a temporary that doesn't exist after this method returns though?

Reply:
The `m_sourceFilename` is the persistent storage in this case.

Reply:
Maybe a PathView would be nicer, but not sure if that'd mean one more include for anyone who wants to use the API.

Reply:
Sorry, my comment was to @AMZN-alexpete 's original suggestion about returning a string_view.

My choice to return a string here was to keep the API consistent with GetSourceFilename() which returns a string& instead of a Path or a PathView. I had considered returning the PathView but I opted for keeping a consistent API instead. I couldn't return a string reference though, because of the temporary string construction needed to extract the extension. I also considered just extracting the extension once and storing it separately, but that felt like overkill.

Reply:
If the string_view works directly with Extension(), I'm good with changing to that. I was just afraid of referencing into a temp string. If that's not what will happen, then great! :)

Reply:
The PathView [decomposition](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/IO/Path/Path.h#L195-L233) functions never any temporary strings and only returns a new PathView object whose string_view member points to a sub-view of the original string.

Reply:
So writing the following code will not return a dangling reference
```c++
return AZ::IO::PathView(m_sourceFilename).Extension().Native();
```

Reply:
I've switched it over to string_view, so that it retains API consistency with GetSourceFilename() but has the improved efficiency of not immediately creating more temporary strings.